js_modules/dagster-ui/packages/ui-core/src/metadata/OverflowingTagCollection.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  Button,
+  ButtonLink,
+  Colors,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Tag,
+} from '@dagster-io/ui-components';
+import {useEffect, useRef, useState} from 'react';
+import styled from 'styled-components';
+
+import {DefinitionTag} from '../graphql/types';
+import {RunTags} from '../runs/RunTags';
+import {buildTagString} from '../ui/tagAsString';
+
+const SHOW_EXTRA_TAGS_WIDTH = 20;
+
+const getOverflowingTag = (target: HTMLDivElement | null) => {
+  if (!target || target.offsetWidth >= target.scrollWidth) {
+    return false;
+  }
+  // find the first tag that is overflowing
+  for (let i = 0; i < target.children.length; i++) {
+    const child = target.children[i];
+    if (
+      child &&
+      child?.getBoundingClientRect()?.right >
+        target.getBoundingClientRect().right - SHOW_EXTRA_TAGS_WIDTH
+    ) {
+      console.log(
+        child,
+        child?.getBoundingClientRect()?.right,
+        target.getBoundingClientRect().right,
+      );
+      return i;
+    }
+  }
+  return target.children.length;
+};
```

Comment:
I feel like this is probably overkill for this -- I think it would be fine to just show a couple tags (maybe threshold of 2) and an overflow link if there are more than that.